### ISSUE TITLE
Narrow to stream lands far in the past, plus marks as read

### ISSUE THREAD

**Issue by gnprice** [zulip/zulip-mobile#2146]
This is a continuation of #1545 -- essentially the same symptoms showed up again for me today, on version 79 which I believe is the latest. Opening separately because we believe we previously fixed at least one issue that could have accounted for #1545. Today's version has an additional, quite serious symptom too.

From the "unreads" view, I tapped a stream. Here's a screenshot of what I got:
![image](https://user-images.githubusercontent.com/28173/37945110-7234e0ca-3133-11e8-8a89-81c6011cc15a.png)
This is a point several months in the past. The stream isn't very busy, so it isn't a huge number of messages behind; in fact, playing around, the actual first unread message starts almost exactly one screenful below the bottom of this screen. (I.e., we're about 2 screenfuls too high.) That might be a clue.

The really serious bit is also evident in the screenshot: there's no mention of unread messages! I believe this means we marked all those messages as read -- when I haven't even seen them yet.

**Comment by gnprice** [comment]
Here's another example, in two screenshots. Again I tapped on a stream from the unreads view. Again, I landed far back in the past -- farther this time, just about 5 screenfuls.

This time, the new messages weren't marked as read at first; here's a screenshot right after tapping on the stream to narrow to it:
![image](https://user-images.githubusercontent.com/28173/37945681-2d4b324a-3136-11e8-91e2-ed01d25db5c3.png)

Immediately after I began scrolling down, though, the messages apparently got marked as read:
![image](https://user-images.githubusercontent.com/28173/37945767-a01d706c-3136-11e8-9475-974414fc07c6.png)
